core/metacling/src/TClingMethodInfo.cxx

```diff
@@ -484,10 +430,13 @@ long TClingMethodInfo::Property() const
    }
    long property = 0L;
    property |= kIsCompiled;
-   const clang::FunctionDecl *fd = GetMethodDecl();
+   const clang::FunctionDecl *fd = GetTargetFunctionDecl();
    if (fd->isConstexpr())
       property |= kIsConstexpr;
-   switch (fd->getAccess()) {
+   // NOTE: this uses `GetDecl()`, to capture the access of the UsingShadowDecl,
+   // which is defined in the derived class and might differ from the access of fd
+   // in the base class.
+   switch (GetDecl()->getAccess()) {
```

Comment:
It feels weird to have to get it twice or are the result of GetDecl and GetTargetFunctionDecl "unrelated"?

Reply:
Yes, and there's a `NOTE` above which explains that? :thinking:

Reply:
Ok.  So they are related but getting from `GetDecl` to `GetTargetFunctionDecl` is non-trivial (conditional loop over using statement) ... and GetDecl is "fast" (even-though it something calls `GetDeclSlow`), so it is okay to call it twice ...

right?

Reply:
Yes:

iter              | `GetDecl()` | `GetTargetFunctionDecl()`
----------------|----------------|-------------------------------
`FunctionDecl` | iter | iter
`UsingShadowDecl` | iter | `iter->getTargetDecl()`

And "Slow" is indeed not *that* slow.